src/Unity.Mathematics/Geometry/MinMaxAABB.cs

```diff
@@ -1,49 +1,331 @@
+        /// </summary>
+        /// <param name="aabb">AABB to test.</param>
+        /// <returns>True if input AABB overlaps with this AABB.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool Overlaps(MinMaxAABB aabb)
         {
-            get { return this.Equals(Empty); }
+            return math.all(Max >= aabb.Min & Min <= aabb.Max);
         }
 
-        public static MinMaxAABB Empty
+        /// <summary>
+        /// Expands the AABB by the given signed distance.
+        /// </summary>
+        /// <remarks>
+        /// Positive distance expands the AABB while negative distance shrinks the AABB.
+        /// </remarks>
+        /// <param name="signedDistance">Signed distance to expand the AABB with.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Expand(float signedDistance)
         {
-            get { return new MinMaxAABB { Min = float3(float.PositiveInfinity), Max = float3(float.NegativeInfinity) }; }
+            Min -= signedDistance;
+            Max += signedDistance;
         }
 
+        /// <summary>
+        /// Encapsulates the given AABB.
+        /// </summary>
+        /// <remarks>
+        /// Modifies this AABB so that it contains the given AABB. If the given AABB is already contained by this AABB,
+        /// then this AABB doesn't change.
+        /// </remarks>
+        /// <seealso cref="Contains(Unity.Mathematics.Geometry.MinMaxAABB)"/>
+        /// <param name="aabb">AABB to encapsulate.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Encapsulate(MinMaxAABB aabb)
         {
             Min = math.min(Min, aabb.Min);
             Max = math.max(Max, aabb.Max);
         }
 
+        /// <summary>
+        /// Encapsulate the given point.
+        /// </summary>
+        /// <remarks>
+        /// Modifies this AABB so that it contains the given point. If the given point is already contained by this AABB,
+        /// then this AABB doesn't change.
+        /// </remarks>
+        /// <seealso cref="Contains(Unity.Mathematics.float3)"/>
+        /// <param name="point">Point to encapsulate.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Encapsulate(float3 point)
         {
             Min = math.min(Min, point);
             Max = math.max(Max, point);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static implicit operator MinMaxAABB(AABB aabb)
         {
             return new MinMaxAABB {Min = aabb.Center - aabb.Extents, Max = aabb.Center + aabb.Extents};
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static implicit operator AABB(MinMaxAABB aabb)
         {
-            return new AABB { Center = (aabb.Min + aabb.Max) * 0.5F, Extents = (aabb.Max - aabb.Min) * 0.5F};
+            return new AABB { Center = aabb.Center, Extents = aabb.HalfExtents };
```

Comment:
Is this a bug, or do the two classes have different definitions of `Extents`?

Reply:
Never mind, looks like the latter. Might be worth a comment explaining...

Reply:
AABB has a different definition of `Extents`.  I'll add a comment to explain.

Reply:
Looks good, thanks

Reply:
Sorry, but we have the opportunity here to avoid creating inconsistent API and we should take it. The proper fix is for this to adopt the same naming scheme as `AABB` or to deprecate and rename the fields / accessors on `AABB`. Having two so similar types both called `AABB` with different meanings for `Size` and `Extent` doesn't sound like a good idea at all.

Reply:
Actually, we don't even have to deprecate anything on AABB, it's also internal. But they must match.

Reply:
I can change it, the main reason why I haven't is that I'm planning on removing AABB entirely and only have the MinMaxAABB.  It just hasn't been "officially" decided but that's the direction I'm heading toward.

Reply:
OK, so if the AABB never goes public and will be removed soon, I'm fine with it.